### ISSUE TITLE
load mp4 file with opencv in Chaquo

### ISSUE THREAD

**Issue by kishcs** [chaquo/chaquopy#140]
HI,

I am not able to load/open video file with opencv. Please check my code below:

```
import cv2
com = cv2.VideoCapture('/storage/emulated/0/sample.mp4')
com.isOpened()

```
**com.isOpened() returns False**

```
import cv2
com = cv2.VideoCapture('/storage/emulated/0/sample.mp4')
ret, frame = com.read()
```
**ret is False and frame is null**

File /storage/emulated/0/sample.mp4 exists
I am able to load image from same path.

```
frame = cv2.imread("/storage/emulated/0/Download/sample.jpg")
print(frame.shape)
```
I am able to print shape as (400, 600, 3)
But not able to open video file.

Please help.
Thanks.

**Comment by mhsmith** [comment]
Unfortunately loading video files into OpenCV requires FFmpeg (#143), which we don't currently provide. If that changes, I'll make sure to update this issue.

You might be able to work around this by reading the video in Java/Kotlin using the Android API, and passing the frames to Python.

**Comment by kishcs** [comment]
@mhsmith 

Thanks for your reply.
Kindly update the issue when **FFmpeg** is supported in future.

Currently we are already following the workaround, which you have suggested. Thanks.

**Comment by chaotianjiao** [comment]
@kishcs  Hi, How do you pass the frames to Python? Can you give some examples?
Acturally, I use opencv-android-sdk to get a camera frame,like this:
`
public void onCameraViewStarted(int width, int height) {
        mRgba = new Mat(height, width, CvType.CV_8UC4);
    }

    public void onCameraViewStopped() {
        mRgba.release();
    }

    public Mat onCameraFrame(CameraBridgeViewBase.CvCameraViewFrame inputFrame) {
        return inputFrame.rgba() ;
    }
`
And I have a **test.py** like this:
`
def test_video(path):
 cap = cv2.VideoCapture(video_path)
    while cap.isOpened():
        ret, frame = cap.read()
        start = timer()
        pro_frame = process(frame, size=4)
        end = timer()
        # cv2.imshow('frame', pro_frame)
    #     if cv2.waitKey(1) & 0xFF == ord('q'):
`
If we can't use cv2.VideoCapture,cv2.imshow(), How can the java frame passing to test_video() ?
If the previous step success, How can you show the result in an Android app(Using SurfaceView or sth else?)

 I will be very appreciated if you can answer my questions above.

**Comment by mhsmith** [comment]
@chaotianjiao: It sounds like you're trying to access the camera rather than reading a video file, so I've created a separate issue #303.

**Comment by sarthaksharma058439** [comment]
Hey i found this video it helps me alot in install and import package in android studio with chaquopy , hopefully this will helps you 

https://youtu.be/BDmB3EFy0Qg

**Comment by abdelhadyElsakka** [comment]
im making object tracking app by aruco so I need to give video to python and return only array 
is there a way for that ?

**Comment by mhsmith** [comment]
I don't know how difficult it is to decode a video in Java/Kotlin using the Android API, but if you can do that, then you can pass the frames to Python as individual images as shown in the example at https://github.com/chaquo/chaquopy/issues/303#issuecomment-637596980.

Alternatively, you could try using mobile-ffmpeg, as described in #143.

**Comment by abdelhadyElsakka** [comment]
can i send the path of the frames directly from the mobile to python ?

**Comment by mhsmith** [comment]
Yes, if you've saved a frame to an image file, you could send the path of that file.
